Bottom: 47999505453068497e99c79755b32fa3f5a97e1a
Top:    ea4844c237cc0f6ab6006fc361a6cb0fad1b3d0c
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-05-30 15:59:24 +0800

vhost_net: introduce tx_can_batch()

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 6be4bea..1e6163f 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -522,6 +522,12 @@ static int get_tx_bufs(struct vhost_net *net,
 	return ret;
 }
 
+static bool tx_can_batch(struct vhost_virtqueue *vq, size_t total_len)
+{
+	return total_len < VHOST_NET_WEIGHT &&
+	       !vhost_vq_avail_empty(vq->dev, vq);
+}
+
 /* Expects to be always run from workqueue - which acts as
  * read-size critical section for our kind of RCU. */
 static void handle_tx(struct vhost_net *net)
@@ -599,8 +605,7 @@ static void handle_tx(struct vhost_net *net)
 			ubufs = NULL;
 		}
 		total_len += len;
-		if (total_len < VHOST_NET_WEIGHT &&
-		    !vhost_vq_avail_empty(&net->dev, vq) &&
+		if (tx_can_batch(vq, total_len) &&
 		    likely(!vhost_exceeds_maxpend(net))) {
 			msg.msg_flags |= MSG_MORE;
 		} else {
